docs: DOC-1773: Nginx Security Advisory (#6235)
spectrocloud/librarium: docs/docs-content/security-bulletins/security-advisories/security-advisories.md

---
sidebar_label: "Security Advisories"
title: "Security Advisories"
description: "Palette Security Advisories for Common Vulnerabilities and Exposures (CVEs)."

> icon: ""

Reviewer: ```suggestion
icon: ""
toc_max_heading_level: 3
```

Reviewer: looks good to me. i think we should increase the heading level on this page so that users can navigate to the related remediation steps more easily

Author: Great catch!

hide_table_of_contents: false
tags: ["security", "cve", "advisories"]
---

> 

Author: Done

> ## NGINX Vulnerability

Author: Good call

- **Release Date**: 03/27/2025
- **Last Updated**: 03/27/2025
- **Severity**: 9.8
- **Affected Versions**: All versions prior to v1.11.0, v1.11.0 - v1.11.4, and v1.12.0
- **Fixed Versions**: v1.11.5 and v1.12.1

### Related CVEs

- [CVE-2025-1097](https://github.com/kubernetes/kubernetes/issues/131007)
- [CVE-2025-1098](https://github.com/kubernetes/kubernetes/issues/131008)
- [CVE-2025-1974](https://github.com/kubernetes/kubernetes/issues/131009)
- [CVE-2025-24513](https://github.com/kubernetes/kubernetes/issues/131005)
- [CVE-2025-24514](https://github.com/kubernetes/kubernetes/issues/131006)

### Timeline

- 03/24/2025: First notified
- 03/24/2025: CVE bulletin published
- 03/26/2025: New Nginx pack published
- 03/26/2025: All managed Palette instances patched

> - 03/27/2025: Manual patch procedure provided for self-hosted Palette for connected and airgapped installations

Reviewer 2: ```suggestion
- 03/27/2025: Manual patch procedure provided for connected and airgapped self-hosted Palette installations
```

> ETA for patch release with the fix for Palette/VerteX 4.6/4.5/4.4:

Reviewer 2: ```suggestion
Estimated release date for patch release with the fix for Palette/VerteX 4.6/4.5/4.4:
```
maybe?

- 03/28/2025, 10:00 PM PST: Palette/Vertex Connected
- 04/02/2025, 10:00 PM PST: Palette/Vertex Airgapped

### Summary

> The identified CVEs affect all ingress-nginx controller deployments using the vulnerable image versions mentioned above.

Reviewer 2: i would add the summary as the first section of the advisory, and then the lines with the affected versions etc. Because when we first read "affected versions", we do not fully understand what we are discussing.

Reviewer 2: Or maybe let's just add one intro sentence after the NGNIX heading saying that this affects nginx and keep the rest as is

Reviewer 2: Here it is:  This advisory outlines security vulnerabilities related to [ingress-nginx](https://github.com/kubernetes/ingress-nginx) and the recommended remediation actions.

When chained together, the vulnerabilities can enable unauthenticated users to execute arbitrary code and download

> confidential information such as secrets available in the cluster. Palette and VerteX use the vulnerable images in the
> management plane. Spectro Cloud also provides the Nginx pack to customers for their workload clusters.

Reviewer 2: ```suggestion
confidential information such as secrets available in the cluster. Palette and VerteX instances use the Ngninx images in the
management plane. Additionally, a Nginx pack is also provided to customers for their workload clusters.
```
maybe?

> As of March 26, 2025, all managed Palette instances have been patched. Customers should follow the below procedure to
> patch self-hosted Palette and VerteX deployments and use the newly published Nginx pack (`1.11.5`) to update affected

Reviewer 2: ```suggestion
As of March 26, 2025, all managed Palette instances have been patched. Customers should follow the below procedure to
manually upgrade the affected component in self-hosted Palette and VerteX deployments and use the newly published Nginx pack (`1.11.5`) to update affected
```

workload clusters.

### Recommended Actions

This vulnerability affects both workload clusters and Palette deployments.

- If you have any workload clusters using the affected version of the Nginx pack, you must update the cluster profile to
  use version `1.11.5` of the Nginx pack. Refer to the
  [Update a Cluster Profile](../../profiles/cluster-profiles/modify-cluster-profiles/update-cluster-profile.md) guide
  for instructions on how to update a cluster profile and apply the updates to workload clusters.

- If you have any self-hosted instances of Palette or VerteX with the affected version of the `ingress-nginx-controller`

>   DaemonSet, you must update it to version `1.11.5`. Follow the steps described below to manually upgrade the

Reviewer 2: ```suggestion
  DaemonSet, you must update it to version `1.11.5`. Follow the steps described in the upcoming sections to manually upgrade the
```

  controller.

#### Multi-Tenant and Dedicated SaaS Palette

As of March 26, 2025, the `ingress-nginx-controller` DaemonSet used for multi-tenant and dedicated SaaS Palette has been

> upgraded to version `1.11.5`. However, affected host clusters must be identified and updated manually to use version

Reviewer 2: ```suggestion
upgraded to version `1.11.5`. However, affected workload clusters must be identified and updated manually to use version
```
